packages/mobile/src/walletConnect/screens/Sessions.tsx

```diff
@@ -0,0 +1,215 @@
+import ItemSeparator from '@celo/react-components/components/ItemSeparator'
+import colors from '@celo/react-components/styles/colors'
+import fontStyles from '@celo/react-components/styles/fonts'
+import variables from '@celo/react-components/styles/variables'
+import { SessionTypes } from '@walletconnect/types'
+import React, { useCallback, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Image, StyleSheet, Text, useWindowDimensions, View } from 'react-native'
+import { ScrollView, TouchableOpacity } from 'react-native-gesture-handler'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { SceneMap, TabBar, TabView } from 'react-native-tab-view'
+import { useDispatch, useSelector } from 'react-redux'
+import Dialog from 'src/components/Dialog'
+import { Namespaces } from 'src/i18n'
+import DrawerTopBar from 'src/navigator/DrawerTopBar'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { closeSession as closeSessionAction } from 'src/walletConnect/actions'
+import { getPendingRequests, getSessions } from 'src/walletConnect/selectors'
+
+const Sessions = () => {
+  const sessions = useSelector(getSessions)
+  const { t } = useTranslation(Namespaces.walletConnect)
+  const [highlighted, setHighlighted] = useState<SessionTypes.Settled | null>(null)
+  const dispatch = useDispatch()
+
+  const closeSession = useCallback(() => {
+    if (!highlighted) {
+      return
+    }
+
+    dispatch(closeSessionAction(highlighted))
+    setHighlighted(null)
+  }, [highlighted, dispatch])
```

Comment:
Simple handlers like this don't need to be wrapped in useCallback and there's some evidence that it actually hurts overall perf. Recommendations on this are conflicting but I'd personally recommend against it.

Reply:
Interesting, my default is to always useCallback things and only if I'm lazy skip it.

But it's a good point that really unless it's referenced further down in the tree there's no point.